pandas/core/generic.py

```diff
@@ -6393,6 +6398,19 @@ def replace(
         regex=False,
         method="pad",
     ):
+        if not (
+            is_scalar(to_replace)
+            or isinstance(to_replace, pd.Series)
+            or is_re_compilable(to_replace)
+            or is_list_like(to_replace)
+            or is_dict_like(to_replace)
```

Comment:
`is_list_like` actually will cover the dict case, so don't need this line

Reply:
Removed, and thank you for the PR feedback!